docs/sso_role_matcher.md

```diff
@@ -0,0 +1,53 @@
+# SSO Role Matcher
+
+Maps configuration for an AWS SSO managed IAM Role to a Kubernetes username and groups.
+
+## Feature state
+
+Alpha
+
+## Use case
+
+Easy and robust configuration for AWS SSO managed roles, which currently have two main issues:
+
+- Confusing configuration. To use an SSO role, a user needs to map the Role ARN of the SSO ROle, minus the path.
```

Comment:
Super small nit, I'd prefer a numbered list or just `First...`, `Second...` given the list is kind of split up.